googledest/users.go

```diff
@@ -228,15 +232,29 @@ func (g *GoogleUsers) updateUser(
 
 	defer wg.Done()
 
-	newUser := newUserForUpdate(person)
-
 	email := person.Attributes["email"]
 
-	_, err := g.AdminService.Users.Update(email, &newUser).Do()
+	oldUser, err := g.getUser(person.CompareValue)
 	if err != nil {
 		eventLog <- personnel_sync.EventLogItem{
 			Event:   "error",
-			Message: fmt.Sprintf("unable to update %s in Users: %s", email, err.Error())}
+			Message: fmt.Sprintf("unable to get old user %s, %s", email, err.Error())}
+		return
+	}
+
+	newUser, err2 := newUserForUpdate(person, oldUser)
+	if err2 != nil {
+		eventLog <- personnel_sync.EventLogItem{
+			Event:   "error",
+			Message: fmt.Sprintf("unable to prepare update for %s in Users: %s", email, err2.Error())}
+		return
+	}
+
+	_, err3 := g.AdminService.Users.Update(email, &newUser).Do()
```

Comment:
Is it worth numbering these err* variables? If you need to on this line, you can leave the colon off before the equal sign.

Reply:
I've started doing that to avoid simple mistakes from using the wrong variable name. Wouldn't know it until an error actually happens and you're trying to troubleshoot the wrong message.

If they were able to be scoped to the `if` block it wouldn't be a big deal.